### ISSUE TITLE
HAL Pagination improvements

### ISSUE THREAD

**Issue by joeribekker** [maykinmedia/djangorestframework-hal#1]
Can you create a Pagination class that, in addition to current behaviour, will also

1) Add `first` and `last` page-links to `_links`
2) Remove `count` from the body
3) Add headers:
   * `X-Total-Count` total objects in result
   * `X-Pagination-Count` total pages in result
   * `X-Pagination-Page` current page number
   * `X-Pagination-Limit` number of objects per page

   See [here](https://github.com/Haarlem/zaaktypecataloguscomponent/blob/master/src/ztc/api/utils/pagination.py) for ideas.

Basically, this has nothing to do with HAL really but is for API Strategy compliancy: https://github.com/Geonovum/KP-APIs/blob/master/Werkgroep%20API%20strategie/Designrules%20v0.1.md#paginering
